packages/server/src/app.ts

```diff
@@ -110,6 +112,32 @@ module.exports = server.listen(env.PORT || 0, async () => {
     }
   }
 
+  // check and create admin user if required
+  if (
+    env.SELF_HOSTED &&
+    !env.MULTI_TENANCY &&
+    env.BB_ADMIN_USER_EMAIL &&
+    env.BB_ADMIN_USER_PASSWORD
+  ) {
+    const checklist = await getChecklist()
+    if (!checklist?.adminUser?.checked) {
+      try {
+        await createAdminUser(
+          env.BB_ADMIN_USER_EMAIL,
+          env.BB_ADMIN_USER_PASSWORD,
+          "default"
```

Comment:
should use the `getTenantId()` function here, ideally even within the `createAdminUser` function, it will always make sure that the correct tenant ID is provided, such as in self host/non-multitenant environments passing down the `"default"` string. Means that if we want to see anything that uses the tenancy logic we can look for usages of that function and don't need to search for magic strings, consts etc.

Can get function from `import { getTenantId } from "@budibase/backend-core/tenancy"`

Reply:
Just to highlight a different import syntax, to get type support you can use
```js
import { tenancy } from "@budibase/backend-core"
tenancy.getTenantId()
```
The `/backend-core/something` imports use the `module.d.ts` file in server which won't resolve to a real implementation in the IDE

Reply:
although upon removing the module file entry for `declare module "@budibase/backend-core/tenancy"` I do now get support locally, looks like this works because of the new `exports` block in the package.json of backend-core

Reply:
however the actual build will still fail, so I'd say for now it's best to use the backend-core root import to get type support